guides/plugins/plugins/content/cms/add-cms-element-via-admin-sdk.md

```diff
@@ -0,0 +1,352 @@
+
+Let's see how the component loading via `viewRenderer.ts` looks like:
+
+```ts
+import Vue from 'vue';
+import { location } from '@shopware-ag/admin-extension-sdk';
+
+// watch for height changes
+location.startAutoResizer();
+
+// start app views
+const app = new Vue({
+    el: '#app',
+    data() {
+        return { location };
+    },
+    components: {
+        'SwagDailymotionElement':
+            () => import('./views/swag-dailymotion/swag-dailymotion-element'),
+        'SwagDailymotionConfig':
+            () => import('./views/swag-dailymotion/swag-dailymotion-config'),
+        'SwagDailymotionPreview':
+            () => import('./views/swag-dailymotion/swag-dailymotion-preview'),
+    },
+    template: `
+        <SwagDailymotionElement
+            v-if="location.is('swag-dailymotion-element')"
+        ></SwagDailymotionElement>
+        <SwagDailymotionConfig
+            v-else-if="location.is('swag-dailymotion-config')"
+        ></SwagDailymotionConfig>
+        <SwagDailymotionPreview
+            v-else-if="location.is('swag-dailymotion-preview')"
+        ></SwagDailymotionPreview>
+    `,
+});
+```
+
+Really straight forward, isn't it? As you probably know from vanilla VueJS's Options API, we just need to load, register and use our VueJS component to make them work.
+
+What's especially interesting here is the use of the `location` object. This is a main concept of the AdminExtensionSDK, where Shopware provides dedicated `locationIds` to offer you places to inject your templates into. For further information on that, it is recommend to have a look at the [documentation of the AdminExtensionSDK](https://shopware.github.io/admin-extension-sdk/docs/guide/concepts/locations) to learn more about its concepts.
+
+In our case, we'll get our own **auto generated** `locationIds`, depending on the name of our CMS element plus suffixes `-element`, `-config` and `-preview`.
+
+Those will be available after **registering the component**, which we'll do in the following chapter.
+
+## Registering a new element
+
+For this topic we head to `mainCommands.ts`, since the registration of CMS elements is something to be done in a global scope.
+
+```ts
+import { cms } from '@shopware-ag/admin-extension-sdk';
+
+const CMS_ELEMENT_NAME = 'swag-dailymotion';
+const CONSTANTS = {
+    CMS_ELEMENT_NAME,
+    PUBLISHING_KEY: `${CMS_ELEMENT_NAME}__config-element`,
+};
+
+void cms.registerCmsElement({
+    name: CONSTANTS.CMS_ELEMENT_NAME,
+    label: 'Dailymotion video',
+    defaultConfig: {
+        dailyUrl: {
+            source: 'static',
+            value: '',
+        },
+    },
+});
+
+export default CONSTANTS;
+```
+
+At first, you see we're importing the AdminExtensionSDK's cms object, used for `cms.registerCmsElement` to register a new element.
+
+This part is easier than the plugin-only method of creating CMS elements, because the formerly required view names of element, config and preview components are completely handled internally.
```

Comment:
Expecting this approach to replace the plugin-way of doing things I think is unnecessary to mention it here.

Reply:
okay, I'll remove that :)